MoviesforSale/app/src/main/java/com/ezequielc/moviesforsale/ShoppingCartAdapter.java

```diff
@@ -0,0 +1,65 @@
+package com.ezequielc.moviesforsale;
+
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
+
+import java.util.ArrayList;
+
+/**
+ * Created by student on 11/8/16.
+ */
+
+public class ShoppingCartAdapter extends RecyclerView.Adapter<ShoppingCartViewHolder> {
+    private ArrayList<Movies> mListOfMovies;
+    private OnMovieRemovedListener mListener;
+
+    public ShoppingCartAdapter(ArrayList<Movies> movies, OnMovieRemovedListener listener){
+        mListOfMovies = movies;
+        mListener = listener;
+    }
+
+    @Override
+    public ShoppingCartViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        return new ShoppingCartViewHolder(LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.shopping_cart_list_layout, parent, false));
+    }
+
+    @Override
+    public void onBindViewHolder(ShoppingCartViewHolder holder, final int position) {
+
+        // Finalizes the position of each Movie
+        final Movies positionOfMovie = mListOfMovies.get(position);
+
+        holder.mMovieShoppingList.setText(positionOfMovie.getName());
+        holder.mRemoveButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                // Removes Movies from the Shopping Cart
+                ShoppingCart.getInstance().removeMovie(positionOfMovie);
+
+                // Specifies which Movie has been removed to the Shopping Cart
+                Toast.makeText(view.getContext(), positionOfMovie.getName()+" Removed",
+                        Toast.LENGTH_SHORT).show();
+
+                // Updates the list of Movies in the Shopping Cart
+                notifyDataSetChanged();
```

Comment:
Don't call `notifyDataSetChanged()` here because that re-draws _everything_ in the RecyclerView, which is inefficient when only one item actually changed. Also, that way you don't get the default animation of the remaining items sliding up after one is removed.

Instead, call `notifyItemRemoved(holder.getAdapterPosition())`, and use `holder.getAdapterPosition()` rather than just `position` since the code in `onClick()` doesn't run right away, and `position` might change before `onClick()` gets called.